design/httpproxy-conflict.md

```diff
@@ -0,0 +1,110 @@
+| Parent Delegation  | Create Orphaned     | HTTP 404 For requests to `conditions.Prefix`   | Error      |
+| Child Delegated    | Orphaned            | No response since requests can't route         | Warning    |
+
+### Invalid Route
+If a route contains configuration errors or warnings, the following table outlines how Contour will react and what response will be sent.  
+
+#### Route Summary
+
+| Category           | Issue               | Request Response                               | Conditions |
+| ------------------ | ------------------- | ---------------------------------------------- | ---------- |
+| Spec.Route.Conditions | Invalid `prefix` | HTTP 404                                       | Error      |
+| Spec.Route.Conditions | Invalid `header` | HTTP 404                                       | Error      |
+| Spec.Route.TimeoutPolicy | Invalid       | Request routes without timeoutpolicy set       | Warning      |
+| Spec.Route.RetryPolicy | Invalid       | Request routes without retrypolicy set       | Warning      |
+| Spec.Route.HTTPHealthCheckPolicy | Invalid       | Request routes without healthcheckpolicy set       | Warning      |
+| Spec.Route.LoadBalancerPolicy | Invalid       | Request routes without loadbalancerpolicy set       | Warning      |
+| Spec.Route.PathRewritePolicy | Invalid       | HTTP 404       | Error      |
+| Spec.Route.RequestHeadersPolicy | Invalid       | Header Not Configured       | Warning      |
+| Spec.Route.ResponseHeadersPolicy | Invalid       | Header Not Configured       | Warning      |
+
+### Invalid Service
+If a service contains configuration errors or warnings, the following table outlines how Contour will react and what response will be sent.
+
+#### Single Service
+For the case where only a single service is referenced from a `Spec.Route` of an HTTPProxy and has an `Error`, Contour will return a `503` status code and also set an error in the Conditions of the object.
+
+#### Multiple Services
+For the case where multiple services are referenced from a `Spec.Route` of an HTTPProxy and has an `Error`, Contour will set the invalid service weight to be `zero` so that no traffic is routed to the service in error and will also set an error in the Conditions of the object.
+
+#### Service Summary
+
+| Category           | Issue            | Request Response       | Conditions |
+| ------------------ | ---------------- | ---------------------- | ---------- |
+| Spec.Route.Service | Missing Service  | HTTP 503               | Error      |
+| Spec.Route.Service | Invalid Protocol | HTTP 503               | Error      |
+| Spec.Route.Service | Port Mismatch    | HTTP 503               | Error      |
+| Spec.Route.Service.RequestHeadersPolicy | Invalid Header  | Header Not Configured | Warning    |
+| Spec.Route.Service.ResponseHeadersPolicy | Invalid Header  | Header Not Configured | Warning    |
+| Spec.Route.Service.UpstreamValidation | Invalid CACertificate | HTTP 503 | Error    |
```

Comment:
I think 502 (bad gateway) is slightly more apt here.

Reply:
👍